File: problem242.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary<char, int> counts = new Dictionary<char, int>();
        foreach (char letter in s) {
            if (!counts.ContainsKey(letter)) {
                counts.Add(letter, 0);
            }
            counts[letter]++;
        }

        foreach (char letter in t) {
            if (!counts.ContainsKey(letter)) {
                return false;
            }
            counts[letter]--;
        }

        foreach (var kvp in counts) {
            if (kvp.Value != 0) {
                return false;
            }
        }

        return true;
    }
}

File: su19/problem242.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        counts = {}
        for char in s:
            if char not in counts:
                counts[char] = 1
            else:
                counts[char] += 1
        for char in t:
            if char not in counts:
                return False
            else:
                counts[char] -= 1
        for count in counts.values():
            if count != 0:
                return False
        return True
        